cmd/otelcol/flags.go

```diff
@@ -0,0 +1,109 @@
+// Copyright  Splunk, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"flag"
+	"io"
+	"strings"
+
+	"go.opentelemetry.io/collector/service/featuregate"
+)
+
+type flags struct {
+	// Command-line flags that are used by Splunk's distribution of the collector
+	configFlags           *stringArrayValue
+	setFlags              *stringArrayValue
+	gatesList             featuregate.FlagValue
+	helpFlag              bool
+	noConvertConfigFlag   bool
+	versionFlag           bool
+	memBallastSizeMibFlag int
+}
+
+var defaultUndeclaredFlag = -1
+
+func (f *flags) getConfigFlags() []string {
```

Comment:
I don't have a strong opinion on this, but I'm wondering if these two getters can go away and we just use direct field access, since we're using direct field access elsewhere.

Reply:
Sure, we can probably remove them since they're not needed at this point.